src/simsopt/configs/zoo.py

```diff
@@ -187,3 +197,82 @@ def get_w7x_data(Nt_coils=48, Nt_ma=10, ppp=2):
+    [SIMSOPT_INSTALL_DIR]/src/simsopt/configs/QUASR_cache/
+
+    Args:
+        ID (int): the ID of the configuration to download.  A pandas dataframe containing metadata on the devices, including all
+                  valid ID numbers is located at: https://quasr.flatironinstitute.org/QUASR.pkl
+                  The database is navigatable online at https://quasr.flatironinstitute.org/
+                  Alternatively, you can download the latest full set of devices from https://zenodo.org/doi/10.5281/zenodo.10050655
+
+        return_style (string, optional): 'simsopt-style' or 'quasr-style', defaults to 'quasr-style'.
+                      For 'simsopt-style', we return [curves_halfperiod, currents] where the curves lie on a half period, similar to get_ncsx_data().
+                      You can take the currents and apply symmetries to get the full device. NOTE: this does not return the magnetic axis. 
+                      quasr-style: [coils_full_torus, surfaces], returns all coils and all surfaces in the object. 
+        
+        verbose (boolean): if true, additional caching and downloading status messages are printed, otherwise they are not.
+
+        returns: depending on return_style: 
+           simsopt-style: [list of simsopt.geo.Coil objects, list of simsopt.field.Current objects]
+           quasr-style: [list of simsopt.geo.SurfaceXYZTensorFourier objects, list of simsopt.field.Coil objects]
+    """
+    
+    if return_style not in ['simsopt-style', 'quasr-style']:
+        raise ValueError(f"invalid return_style: {return_style}, must be either simsopt-style or quasr-style")
+    
+    id_str = f"{ID:07d}" # string to 7 digits
+    url = f'https://quasr.flatironinstitute.org/simsopt_serials/{id_str[0:4]}/serial{id_str}.json'
+    success = False
+
+    if use_cache:
+        if os.access(THIS_DIR, os.W_OK):
+            FILE_PATH = THIS_DIR / 'QUASR_cache' / f'serial{id_str}.json' # write to cache in simsopt source dir
+        elif os.access(os.getcwd(), os.W_OK): 
+            FILE_PATH =  Path(os.getcwd()) / 'QUASR_cache'  / f"serial{id_str}.json" # create a local cache 
+        else: 
+            raise ValueError("could not find a writeable location, try again with use_cache=False")
+    
+        exists = os.path.exists(FILE_PATH)
+
+        if exists:
+            if verbose:
+                print(f"ID={id_str} is cached, loading...")
+            surfaces, coils = load(FILE_PATH)
+            success = True
+
+
+    if not success: 
+        try:
+            r = requests.get(url)
+        except:
+            raise Exception(f"requests failure on ID {ID:07}")
+
+        if r.status_code == 200:
+            print(f"ID={ID:07} downloaded successfully")
+            surfaces, coils = json.loads(r.content, cls=GSONDecoder)
+            success = True
+            
+            if use_cache: 
+                with open(FILE_PATH, 'wb') as f:
```

Comment:
This will fail if the directory doesn't exist yet.
```suggestion
                os.makedirs(FILE_PATH.parent, exist_ok=True)
                with open(FILE_PATH, 'wb') as f:
```